app/models/zone.rb

```diff
@@ -0,0 +1,63 @@
+class Zone
+  attr_reader :facility
+
+  def initialize(facility)
+    @facility = facility
+  end
+
+  def blood_pressures
+    BloodPressure
+      .includes(encounter: {facility: :facility_group})
+      .where(facilities: {facility_group: facility_group})
+      .where(facilities: {zone: zone})
```

Comment:
Would be good to run some EXPLAIN on this w/ facilities with lots of data, and also just get some comparison numbers between this PR and the production query to retrieve blood_pressures.

Reply:
Yeah, I'll definitely do that. 

But after some thought, I think we need to change the definition of how these BPs are calculated. I'm calculating them using encounters (the same way we do this for FGs currently). But I think this approach will not fly, because it can result in dangling BPs on the device.

If BPs are scoped by encounters, that means they are scoped by the facility of the BP. This is fine if the user is downloading all the patient data for the FG. But if we're syncing only block-level data, we should only sync those patients that the User can see in their block and then scope the BPs by patients on the device, rather than by facility.

If we don't do that, then this can happen:

```
Patient 1 registers in Facility 1
Patient 1 goes to Facility 1 in Block 1 to take BP1
Patient 1 goes to Facility 2 in Block 2 to take BP2
Patient 1 goes to Facility 2 in Block 2 to take BP3
Patient 1 goes to Facility 2 in Block 2 to take BP4

Nurse 1 from Facility 1 would be able to see the patient, and only BP1
Nurse 2 from Facility 2 can't see this patient because they are not registered here, but will sync BPs from BP2 to BP4.
```

This is quite non-ideal (even if the example is a little contrived).

It should instead be,

```
Patient 1 registers in Facility 1
Patient 1 goes to Facility 1 in Block 1 to take BP1
Patient 1 goes to Facility 2 in Block 2 to take BP2
Patient 1 goes to Facility 2 in Block 2 to take BP3
Patient 1 goes to Facility 2 in Block 2 to take BP4

Nurse 1 from Facility 1 should be able to see Patient 1 and all their BPs (regardless of where they were recorded)
Nurse 2 from Facility 2 can't see this patient
```